api/cloudcontroller/ccv3/space_test.go

```diff
@@ -159,4 +160,74 @@ var _ = Describe("Spaces", func() {
 			})
 		})
 	})
+
+	Describe("UpdateSpace", func() {
+		var (
+			spaceToUpdate Space
+			updatedSpace  Space
+			warnings      Warnings
+			executeErr    error
+		)
+
+		JustBeforeEach(func() {
+			updatedSpace, warnings, executeErr = client.UpdateSpace(spaceToUpdate)
+		})
+
+		When("the organization is updated successfully", func() {
+			BeforeEach(func() {
+				response := `{
+					"guid": "some-space-guid",
+					"name": "some-space-name",
+					"metadata": {
+						"labels": {
+							"k1":"v1",
+							"k2":"v2"
+						}
+					}
+				}`
+
+				expectedBody := map[string]interface{}{
+					"name": "some-space-name",
+					"metadata": map[string]interface{}{
+						"labels": map[string]string{
+							"k1": "v1",
+							"k2": "v2",
+						},
+					},
+				}
+
+				server.AppendHandlers(
+					CombineHandlers(
+						VerifyRequest(http.MethodPatch, "/v3/spaces/some-guid"),
+						VerifyJSONRepresenting(expectedBody),
+						RespondWith(http.StatusOK, response, http.Header{"X-Cf-Warnings": {"this is a warning"}}),
+					),
+				)
+
+				spaceToUpdate = Space{
+					Name: "some-space-name",
+					GUID: "some-guid",
+					Metadata: &Metadata{
+						Labels: map[string]types.NullString{
+							"k1": types.NewNullString("v1"),
+							"k2": types.NewNullString("v2"),
+						},
+					},
+				}
+			})
+
+			It("should include the labels in the JSON", func() {
+				Expect(executeErr).ToNot(HaveOccurred())
+				Expect(server.ReceivedRequests()).To(HaveLen(3))
+				Expect(len(warnings)).To(Equal(0))
```

Comment:
I don't understand why `len(warnings)` should be 0. It looks like line 203 has a warning in it.

Reply:
Good catch. Specifying 0 warnings was a coding error, and the implementation had hardwired `nil` for the returned warnings.